### PR TITLE
Render weapon and shield. 

### PR DESCRIPTION
**Pull request by gugus**
You cannot really test this feature now with master as you cannot draw out your weapon (for that, animation layering is needed), but it works (tested it with some hacks to draw the weapon out), and shield are displayed too.

### CODE REVIEW

File: apps/openmw/mwrender/npcanimation.cpp [hunk 1]
```diff
@@ -464,4 +469,23 @@ void NpcAnimation::addPartGroup(int group, int priority, const std::vector<ESM::
...
+void NpcAnimation::showWeapons(bool showWeapon)
+{
+    mShowWeapons = showWeapon;
+    if(showWeapon)
+    {
+        MWWorld::InventoryStore &inv = MWWorld::Class::get(mPtr).getInventoryStore(mPtr);
+        mWeapon = inv.getSlot(MWWorld::InventoryStore::Slot_CarriedRight);
+        if(mWeapon != inv.end()) // special case for weapons
+        {
+            const ESM::Weapon* weapon = (mWeapon->get<ESM::Weapon>())->mBase;
```
Comment (ghost) [inline apps/openmw/mwrender/npcanimation.cpp]: This needs to be changed to handle Lockpicks and Probes too. MWWorld::Class::getModel should be helpful here.

**Comment by zinnschlag** [comment]
From what I see you seem to add some more container iterators to the NpcAnimation class, but you don't update them anywhere (like we are doing all the existing iterators). That will fail badly when a non-player NPC crosses a cell boarder or any NPC changes weapon while the weapon is drawn.

**Comment by gugus** [comment]
It is updated, see the last commit. (showWeapon is called whenever there is an update)

Date: Sun, 28 Apr 2013 05:48:37 -0700

**Comment by zinnschlag** [comment]
I see. Merge incoming ...

**Pull request merged**